lib/src/kratos_client.dart

```diff
@@ -801,55 +832,113 @@ class KratosClient {
   }
 
   Future<RecoveryFlowResult> getRecoveryFlow() async {
-    final recoveryFlow = await _client.get(
-      _buildUri(path: 'self-service/recovery/$_flowType'),
-    );
     try {
-      final decodedResult =
-          jsonDecode(recoveryFlow.body) as Map<String, dynamic>;
-      final dynamic recoveryFlowId = decodedResult['id'];
-      switch (recoveryFlowId) {
-        case String _:
-          return RecoveryFlow(recoveryFlowId);
-        default:
-          return RecoveryFlowError();
-      }
+      final recoveryFlowDto = await _getRecoveryFlow();
+
+      return switch (recoveryFlowDto) {
+        RecoveryFlowDto(:final id) => RecoveryFlow(id),
+        _ => RecoveryFlowError(),
+      };
     } catch (e, st) {
       _logger.warning('Error getting recovery flow', e, st);
       return RecoveryFlowError();
     }
   }
 
+  Future<RecoveryFlowDto?> _getRecoveryFlow({
+    String? flowId,
+    bool addFlowType = true,
+  }) async {
+    try {
+      var path = 'self-service/recovery';
+      if (addFlowType) {
+        path += '/$_flowType';
+      }
+
+      final streamedResponse = await _client.send(
```

Comment:
Why do we need to go through _streaming_ here? What was wrong with the previous approach?

Reply:
It allows for greater request behavior configuration, in this case `followRedirects = false` is set.

Reply:
Ahh, got it, thx!